src/Parser/LASParser.js

```diff
@@ -1,7 +1,24 @@
 import * as THREE from 'three';
 import LASLoader from 'Parser/LASLoader';
+import { spawn, Pool } from 'threads';
 
 const lasLoader = new LASLoader();
+const lasWorkerLoader = {
+    async parseFile(data, options) {
+        if (!this.pool) {
+            const initWorker = () =>
+                new Worker(
+                    /* webpackChunkName: "itowns_lasparser" */
+                    new URL('../Worker/LASWorker.js', import.meta.url),
+                );
+            this.pool = new Pool(async () => spawn(initWorker()), {
+                size: 1,
```

Comment:
Do we really need a pool if size is 1 ? Or should we make size configurable ?

Reply:
I used a pool to experiment with different number of workers but failed to notice a difference in performance when using a size $\geq 2$...

Those results could be explained by the fact that I have a low numbers of CPU-cores and those are surely hardware-dependent.
I don't have enough data (performance tests or analytics) to deduce what would be the perfect size but this could be fined-tune later! =)

At least with a single worker, the main thread is not clogged!

P.S.: Maplibre did some A/B testing on their workers and limit the worker pool size to $1$ (see https://github.com/maplibre/maplibre-gl-js/pull/2354). However this could also be task-dependent.

Reply:
Interresting! Thanks